tests/suites/test_suite_psa_crypto.data

```diff
@@ -1250,173 +1250,509 @@ mac_setup:PSA_KEY_TYPE_HMAC:"000102030405060708":PSA_ALG_HMAC(PSA_ALG_MD5):PSA_E
 PSA MAC: bad order function calls
 mac_bad_order:
 
-PSA MAC sign: RFC4231 Test case 1 - HMAC-SHA-224
+PSA MAC sign hash: RFC4231 Test case 1 - HMAC-SHA-224
```

Comment:
Two points:

1. I'm not a fan of the naming: the operation we do is "PSA MAC sign", regardless of which usage bits we set in the key. There's no operation that is conceptually "PSA MAC sign hash" so we shouldn't call it that.
2. I'm not sure it's useful to have three copies of each test. IMO there are two independent things we want to be testing here: (a) whether the result of the operation is correct (using a number of test vectors) and (b) whether permissions on the key are checked properly. And this is already reflected in the tests: with have the test vectors here for (a), and (b) was handled by `mac_key_policy` (modified in the previous commit), so I don't think it needs to be re-done here.

I suggest replacing this commit with one that would keep the test data unchanged, and in the test function just use the `MESSAGE` policy instead of the `HASH` one. We already have test ensuring that `HASH` gets completed to `HASH|MESSAGE` so `MESSAGE` is effectively the minimal policy to use here, so that's we should use when the testing against test vectors.

Reply:
re 1: the test name is not really describing the operation, but is rather a short summary of what we are testing. I agree it could be a bit more descriptive, something like `PSA MAC sign, only hash flag`
re 2: the fix from first commit is modifying the functions `psa_mac_compute_internal` and `psa_mac_setup`. Here is a short summary of function calls executed by `test_suite_psa_crypto.function`:
```
mac_sign -> psa_mac_compute -> psa_mac_compute_internal
mac_sign -> psa_mac_sign_setup -> psa_mac_setup
mac_key_policy -> psa_mac_sign_setup -> psa_mac_setup
mac_verify -> psa_mac_verify_setup -> psa_mac_setup
```
`psa_mac_compute` is tested only by the `mac_sign`, so I think that the `mac_sign` tests have to be extended to cover the execution of `psa_mac_compute` with both `_MESSAGE` and `_HASH` flags. This behavior cannot be tested by the `mac_key_policy` alone.

Reply:
Fixed test cases descriptions

Reply:
re 1: thanks for improving the names

re 2: good point, I was assuming `psa_mac_compute()` would be implemented by calling `psa_mac_setup()` at some point, but that's evidently not the case due to drivers, my mistake. I fully agree we want to be testing this function too :)

I'm still not happy with triplicating all the `mac_sign` and `mac_verify` test cases, and still think it would be better to test in an orthogonal way: `mac_key_policy` for all things related to policy, and `mac_sign` and `mac_verify` for functional correctness.

I think that can be achieved by extending the `mac_key_policy` function to also test `psa_mac_compute()` and `psa_mac_verify()`. For `psa_mac_compute()` it's easy: use an all-0 (or any other fixed value) buffer as input, a large enough buffer for the output, and the expected status is the same as for `psa_mac_sign_setup()`. For `psa_mac_verify()` we need a bit more care: re-use the input and output buffers from `psa_mac_compute()`, but for the expected status do something like:
```
if( expected_status_sign != PSA_SUCCESS && expected_status_verify == PSA_SUCCESS )
    TEST_EQUAL( status, PSA_ERROR_INVALID_SIGNATURE );
else
    TEST_EQUAL( status, expected_status_verify );
```

Reply:
implemented requested change